player/public/coreplugins/org.ekstep.videorenderer-1.0/renderer/plugin.js

```diff
@@ -37,30 +39,76 @@ org.ekstep.contentrenderer.baseLauncher.extend({
     },
     createVideo: function(path, data) {
         EkstepRendererAPI.dispatchEvent("renderer:splash:hide");
-        video = document.createElement('video');
+        EkstepRendererAPI.dispatchEvent("renderer:content:start");
+
+        video = document.createElement('video-js');
         video.style.width = '100%';
         video.style.height = '100%';
-        video.controls = true;
-        if (window.cordova) video.controlsList = 'nodownload';
-        video.autoplay = true;
-        video.preload = "auto";
-        video.className = 'video-js vjs-default-skin';
-        this.addToGameArea(video);
-        EkstepRendererAPI.dispatchEvent("renderer:content:start");
-        data.mimeType === 'video/x-youtube' ? this._loadYoutube(data.artifactUrl) : this._loadVideo(path);
+        video.style.position = 'absolute';
+        video.id = "videoElement";
+        video.className = 'vjs-default-skin';
+        document.body.appendChild(video);
+
+        //Todo need to remove after testing
+        //data.artifactUrl = "https://sunbirdspikemedia-inct.streaming.media.azure.net/afcc5a99-d0c4-4ef5-9dfe-dc403a1269fb/learn-colors-with-numbers-in-kid.ism/manifest(format=m3u8-aapl-v3)";
+
+        if(data.mimeType === "video/x-youtube"){
+            this._loadYoutube(data.artifactUrl);
+        }else if(this.isStreaming && data.mimeType != "video/x-youtube"){
+            data.mimeType = this.supportedStreamingMimeType;
+            this._loadVideo(data.artifactUrl, data);
+        }else{
+            this._loadVideo(path, data);
+        }
+        
         video.onvolumechange = function() {
             (video.muted) ? EkstepRendererAPI.dispatchEvent('renderer:overlay:mute') : EkstepRendererAPI.dispatchEvent('renderer:overlay:unmute');
         };
         $("video").bind("contextmenu",function() {
             return false;
         });
-    },
-    _loadVideo: function(path) {
+    },    
+    _loadVideo: function(path, data) {
+        var instance = this;
+        if (instance.isStreaming && !navigator.onLine) {
+            EkstepRendererAPI.logErrorEvent('No internet', {
+                'type': 'content',
+                'action': 'play',
+                'severity': 'error'
+            });
+            instance.throwError({message:'Internet not available. Please connect and try again.'});
+        }
```

Comment:
break the function execution, as noInternet not need to execute any further lines for streaming.